### ISSUE TITLE
multi-class segmentation discuss

### ISSUE THREAD

**Issue by smallboy-code** [JuliaWolleb/Diffusion-based-Segmentation#54]
Hi @JuliaWolleb , thank you for your great work. I want to solve the multi-class segmentation problem and I delete the "label = torch.where(label > 0, 1, 0).float()". Now, I got some output looks not well as follows:
![2](https://github.com/JuliaWolleb/Diffusion-based-Segmentation/assets/61675340/d1f3a5cb-d686-4373-89e9-69f48261768b)
![3](https://github.com/JuliaWolleb/Diffusion-based-Segmentation/assets/61675340/936e67aa-883d-41e6-a76b-a0da6dd1e17b)
![5](https://github.com/JuliaWolleb/Diffusion-based-Segmentation/assets/61675340/105d0064-b84e-4c54-a751-177b58028235)
![6](https://github.com/JuliaWolleb/Diffusion-based-Segmentation/assets/61675340/f71349dc-2b12-4615-bc14-c29374dbee6f)
![8](https://github.com/JuliaWolleb/Diffusion-based-Segmentation/assets/61675340/00ac66f8-2688-4fdf-8b36-23e45deadc8b)
![7](https://github.com/JuliaWolleb/Diffusion-based-Segmentation/assets/61675340/cb03b4f6-64d2-4f71-87ac-8f1825c8805b)

So, can you give me some advices about this problem?

**Comment by JuliaWolleb** [comment]
Hi
How did you train this multi-class approach? It seems like you trained on a segmentation mask with a single channel, but different values (0,1,2,3) for the different classes?
I suggest you train with one-hot-encoding, i.e., one binary mask for each of the classes.

**Comment by smallboy-code** [comment]
Yes, I trained on a segmentation mask with a single channel with different values (0,1,2,3) for the different classes. The one binary mask for each of the classes means training on a segmentation mask with three channels with WT,TC and ET?

**Comment by JuliaWolleb** [comment]
yes exactly, all three classes will have a separate binary channel each.

**Comment by smallboy-code** [comment]
Yes, I have also try to do this, but I found the finnal output will be like this:
![图片1](https://github.com/JuliaWolleb/Diffusion-based-Segmentation/assets/61675340/6f6b32fd-1411-45f1-a7cc-691099741d15)
WT,TC,ET,respectively. I think the results are worse than one channel.

**Comment by ZhengChen6** [comment]
> Yes, I have also try to do this, but I found the finnal output will be like this: ![图片1](https://private-user-images.githubusercontent.com/61675340/280683949-6f6b32fd-1411-45f1-a7cc-691099741d15.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA2NTg0MDEsIm5iZiI6MTcxMDY1ODEwMSwicGF0aCI6Ii82MTY3NTM0MC8yODA2ODM5NDktNmY2YjMyZmQtMTQxMS00NWYxLWE3Y2MtNjkxMDk5NzQxZDE1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzE3VDA2NDgyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZhZGMzMDMxNjA5YjJmOWY5NmE1ZTQ5MDVlNDcwN2UwM2Q5MDNiMmI4NDk1MTdkMzRiY2M4ZGU4MWM5YTY5YzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.bAi43tppin7IvpBtyDBscGCOnGmVVuY1secQoZkQ1Ds) WT,TC,ET,respectively. I think the results are worse than one channel.

Hi, I also want to use multi-class segmentation for training, have you solved this problem? @smallboy-code

**Comment by smallboy-code** [comment]
It remains unresolved. @ZhengChen6
